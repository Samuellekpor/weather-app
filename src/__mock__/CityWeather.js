import React from 'react';
import {
  CircularProgressbarWithChildren,
  buildStyles,
} from 'react-circular-progressbar';
import RadialSeparators from '../components/RadialSeparators';
import 'react-circular-progressbar/dist/styles.css';

function CityWeather() {
  const state = [
    {
      id: 500,
      main: 'Rain',
      description: 'light rain',
      temperature: '23.98',
      visibility: 10000,
      humidity: 100,
      icon: '10n',
    },
  ];

  return (
    <div>
      {state.length > 0 && (
        <div className="weatherDetails" key={state[0].id}>
          <div className="detTitle">
            <h4 className="cityName">{state[0].name}</h4>
            <p className="temp">{`${state[0].temperature} Â°C`}</p>
          </div>
          <img className="weatherIcon" src={`https://openweathermap.org/img/wn/${state[0].icon}@2x.png`} alt="/" />
          <p className="mainWeather">
            {state[0].main}
            :
            {' '}
            {state[0].description}
          </p>
          <div className="humidity">
            <h4 className="humTitle">Humidity:</h4>
            <div className="humBar">
              <CircularProgressbarWithChildren
                value={state[0].humidity}
                text={`${state[0].humidity}%`}
                strokeWidth={10}
                styles={buildStyles({
                  strokeLinecap: 'butt',
                  pathColor: '#3f63a6',
                  width: '20px',
                  height: '20px',
                })}
              >
                <RadialSeparators
                  count={12}
                  style={{
                    background: '#fff',
                    width: '2px',
                    // This needs to be equal to props.strokeWidth
                    height: `${10}%`,
                  }}
                />
              </CircularProgressbarWithChildren>
            </div>
          </div>
          <div className="visibility">
            <h4 className="visTitle">Visibility:</h4>
            <span className="visibilityDetail">
              {state[0].visibility}
              {' km'}
            </span>
          </div>
        </div>
      )}
    </div>
  );
}

export default CityWeather;
