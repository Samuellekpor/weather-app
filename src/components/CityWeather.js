import React, { useEffect, useRef } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import {
  CircularProgressbarWithChildren,
  buildStyles,
} from 'react-circular-progressbar';
import RadialSeparators from './RadialSeparators';
import 'react-circular-progressbar/dist/styles.css';
import getCityWeatherFromApi from '../redux/city-weather';

function CityWeather() {
  const params = useParams();
  const dispatch = useDispatch();
  const state = useSelector((state) => state.cities.cityWeather);
  const { name } = params;
  const effectRan = useRef(false);

  useEffect(() => {
    if (effectRan.current || process.env.NODE_ENV !== 'development') {
      dispatch(getCityWeatherFromApi(name));
    }
    effectRan.current = true;
  }, [name, dispatch]);

  return (
    <div>
      {state.length > 0 && (
        <div className="weatherDetails" key={state[0].id}>
          <div className="detTitle">
            <h4 className="cityName">{name}</h4>
            <p className="temp">{`${state[0].temperature} Â°C`}</p>
          </div>
          <img className="weatherIcon" src={`https://openweathermap.org/img/wn/${state[0].icon}@2x.png`} alt="/" />
          <p className="mainWeather">
            {state[0].main}
            :
            {' '}
            {state[0].description}
          </p>
          <div className="humidity">
            <h4 className="humTitle">Humidity:</h4>
            <div className="humBar">
              <CircularProgressbarWithChildren
                value={state[0].humidity}
                text={`${state[0].humidity}%`}
                strokeWidth={10}
                styles={buildStyles({
                  strokeLinecap: 'butt',
                  pathColor: '#3f63a6',
                  width: '20px',
                  height: '20px',
                })}
              >
                <RadialSeparators
                  count={12}
                  style={{
                    background: '#fff',
                    width: '2px',
                    // This needs to be equal to props.strokeWidth
                    height: `${10}%`,
                  }}
                />
              </CircularProgressbarWithChildren>
            </div>
          </div>
          <div className="visibility">
            <h4 className="visTitle">Visibility:</h4>
            <span className="visibilityDetail">
              {state[0].visibility}
              {' km'}
            </span>
          </div>
        </div>
      )}
    </div>
  );
}

export default CityWeather;
