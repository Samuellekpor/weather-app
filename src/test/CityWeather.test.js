import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import CityWeather from '../__mock__/CityWeather';
import store from '../redux/store';

it('City component should render correctly', () => {
  const cityWeather = render(
    <Provider store={store}>
      <CityWeather />
    </Provider>,
  );
  expect(cityWeather).toMatchSnapshot();
});
