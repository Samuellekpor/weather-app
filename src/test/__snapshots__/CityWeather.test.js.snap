// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="weatherDetails"
        >
          <div
            class="detTitle"
          >
            <h4
              class="cityName"
            />
            <p
              class="temp"
            >
              23.98 °C
            </p>
          </div>
          <img
            alt="/"
            class="weatherIcon"
            src="https://openweathermap.org/img/wn/10n@2x.png"
          />
          <p
            class="mainWeather"
          >
            Rain
            :
             
            light rain
          </p>
          <div
            class="humidity"
          >
            <h4
              class="humTitle"
            >
              Humidity:
            </h4>
            <div
              class="humBar"
            >
              <div
                data-test-id="CircularProgressbarWithChildren"
              >
                <div
                  style="position: relative; width: 100%; height: 100%;"
                >
                  <svg
                    class="CircularProgressbar "
                    data-test-id="CircularProgressbar"
                    viewBox="0 0 100 100"
                  >
                    <path
                      class="CircularProgressbar-trail"
                      d="
      M 50,50
      m 0,-45
      a 45,45 0 1 1 0,90
      a 45,45 0 1 1 0,-90
    "
                      fill-opacity="0"
                      stroke-width="10"
                      style="stroke-linecap: butt; stroke-dasharray: 282.7433388230814px 282.7433388230814px; stroke-dashoffset: 0px;"
                    />
                    <path
                      class="CircularProgressbar-path"
                      d="
      M 50,50
      m 0,-45
      a 45,45 0 1 1 0,90
      a 45,45 0 1 1 0,-90
    "
                      fill-opacity="0"
                      stroke-width="10"
                      style="stroke: #3f63a6; stroke-linecap: butt; stroke-dasharray: 282.7433388230814px 282.7433388230814px; stroke-dashoffset: 0px;"
                    />
                    <text
                      class="CircularProgressbar-text"
                      x="50"
                      y="50"
                    >
                      100%
                    </text>
                  </svg>
                  <div
                    data-test-id="CircularProgressbarWithChildren__children"
                    style="position: absolute; width: 100%; height: 100%; margin-top: -100%; display: flex; flex-direction: column; justify-content: center; align-items: center;"
                  >
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.08333333333333333turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.16666666666666666turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.25turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.3333333333333333turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.41666666666666663turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.5turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.5833333333333333turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.6666666666666666turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.75turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.8333333333333333turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                    <div
                      style="position: absolute; height: 100%; transform: rotate(0.9166666666666666turn);"
                    >
                      <div
                        style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="visibility"
          >
            <h4
              class="visTitle"
            >
              Visibility:
            </h4>
            <span
              class="visibilityDetail"
            >
              10000
               km
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="weatherDetails"
      >
        <div
          class="detTitle"
        >
          <h4
            class="cityName"
          />
          <p
            class="temp"
          >
            23.98 °C
          </p>
        </div>
        <img
          alt="/"
          class="weatherIcon"
          src="https://openweathermap.org/img/wn/10n@2x.png"
        />
        <p
          class="mainWeather"
        >
          Rain
          :
           
          light rain
        </p>
        <div
          class="humidity"
        >
          <h4
            class="humTitle"
          >
            Humidity:
          </h4>
          <div
            class="humBar"
          >
            <div
              data-test-id="CircularProgressbarWithChildren"
            >
              <div
                style="position: relative; width: 100%; height: 100%;"
              >
                <svg
                  class="CircularProgressbar "
                  data-test-id="CircularProgressbar"
                  viewBox="0 0 100 100"
                >
                  <path
                    class="CircularProgressbar-trail"
                    d="
      M 50,50
      m 0,-45
      a 45,45 0 1 1 0,90
      a 45,45 0 1 1 0,-90
    "
                    fill-opacity="0"
                    stroke-width="10"
                    style="stroke-linecap: butt; stroke-dasharray: 282.7433388230814px 282.7433388230814px; stroke-dashoffset: 0px;"
                  />
                  <path
                    class="CircularProgressbar-path"
                    d="
      M 50,50
      m 0,-45
      a 45,45 0 1 1 0,90
      a 45,45 0 1 1 0,-90
    "
                    fill-opacity="0"
                    stroke-width="10"
                    style="stroke: #3f63a6; stroke-linecap: butt; stroke-dasharray: 282.7433388230814px 282.7433388230814px; stroke-dashoffset: 0px;"
                  />
                  <text
                    class="CircularProgressbar-text"
                    x="50"
                    y="50"
                  >
                    100%
                  </text>
                </svg>
                <div
                  data-test-id="CircularProgressbarWithChildren__children"
                  style="position: absolute; width: 100%; height: 100%; margin-top: -100%; display: flex; flex-direction: column; justify-content: center; align-items: center;"
                >
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.08333333333333333turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.16666666666666666turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.25turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.3333333333333333turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.41666666666666663turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.5turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.5833333333333333turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.6666666666666666turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.75turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.8333333333333333turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                  <div
                    style="position: absolute; height: 100%; transform: rotate(0.9166666666666666turn);"
                  >
                    <div
                      style="background: rgb(255, 255, 255); width: 2px; height: 10%;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="visibility"
        >
          <h4
            class="visTitle"
          >
            Visibility:
          </h4>
          <span
            class="visibilityDetail"
          >
            10000
             km
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
